Using empty password in liquibase Maven plugin causes: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because "configuredValue"

### Search first

- [x] I searched and no similar issues were found

### Description

When we use an empty password for database connection, "update" goal of Maven plugin fails with error:

> [ERROR] Failed to execute goal org.liquibase:liquibase-maven-plugin:4.30.0:update (migrate-dev-db) on project db: Cannot invoke "String.equals(Object)" because "configuredValue" is null -> [Help 1]

The issue appears from the version 4.30.0.

### Steps To Reproduce

1. Use the properties in pom.xml (mind empty `localdev.jdbc.password`):
```
        <localdev.jdbc.driver>org.postgresql.Driver</localdev.jdbc.driver>
        <localdev.jdbc.url>jdbc:postgresql://localhost:5432</localdev.jdbc.url>
        <localdev.jdbc.user>postgres</localdev.jdbc.user>
        <localdev.jdbc.password></localdev.jdbc.password>
```

2. Configure the plugin with:
```
            <plugin>
                <groupId>org.liquibase</groupId>
                <artifactId>liquibase-maven-plugin</artifactId>
                <version>${liquibase.version}</version>
                <executions>
                    <execution>
                        <id>migrate-test-db</id>
                        <goals>
                            <goal>update</goal>
                        </goals>
                        <phase>generate-sources</phase>
                        <configuration>
                            <url>${localdev.jdbc.url}/${localdev.jdbc.testdb}</url>
                        </configuration>
                    </execution>
                </executions>
                <configuration>
                    <username>${localdev.jdbc.user}</username>
                    <password>${localdev.jdbc.password}</password>
                    <verbose>true</verbose>
                    <changeLogFile>liquibase/changelog-main.xml</changeLogFile>
                </configuration>
            </plugin>
```

3. Run the goal and get:
```
[ERROR] Failed to execute goal org.liquibase:liquibase-maven-plugin:4.30.0:update (migrate-dev-db) on project db: Cannot invoke "String.equals(Object)" because "configuredValue" is null -> [Help 1]
```



### Expected/Desired Behavior

The empty password should be expected as per code comments/documentation:

1. https://github.com/liquibase/liquibase/blob/28ab1facd7357c912f9ab453923201b3b020138f/liquibase-maven-plugin/src/main/java/org/liquibase/maven/plugins/AbstractLiquibaseMojo.java#L149
2. https://docs.liquibase.com/tools-integrations/maven/commands/maven-update.html --> search for "empty password"


### Liquibase Version

4.30.0

### Database Vendor & Version

postgresql stable 14.13

### Liquibase Integration

maven

### Liquibase Extensions

liquibase-nochangeloglock

### OS and/or Infrastructure Type/Provider

Mac OSX

### Additional Context

The full stack trace of the error:
```
[ERROR] Failed to execute goal org.liquibase:liquibase-maven-plugin:4.30.0:update (migrate-dev-db) on project db: Cannot invoke "String.equals(Object)" because "configuredValue" is null -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.liquibase:liquibase-maven-plugin:4.30.0:update (migrate-dev-db) on project db: Cannot invoke "String.equals(Object)" because "configuredValue" is null
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: Cannot invoke "String.equals(Object)" because "configuredValue" is null
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.execute (AbstractLiquibaseMojo.java:1062)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because "configuredValue" is null
    at liquibase.configuration.ConfiguredValueModifierFactory.override (ConfiguredValueModifierFactory.java:43)
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.lambda$modifyValue$28 (AbstractLiquibaseMojo.java:1513)
    at liquibase.Scope.child (Scope.java:203)
    at liquibase.Scope.child (Scope.java:179)
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.modifyValue (AbstractLiquibaseMojo.java:1513)
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.setPassword (AbstractLiquibaseMojo.java:139)
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.lambda$execute$2 (AbstractLiquibaseMojo.java:979)
    at liquibase.Scope.lambda$child$0 (Scope.java:194)
    at liquibase.Scope.child (Scope.java:203)
    at liquibase.Scope.child (Scope.java:193)
    at liquibase.Scope.child (Scope.java:172)
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.lambda$execute$3 (AbstractLiquibaseMojo.java:955)
    at liquibase.Scope.lambda$child$0 (Scope.java:194)
    at liquibase.Scope.child (Scope.java:203)
    at liquibase.Scope.child (Scope.java:193)
    at liquibase.Scope.child (Scope.java:172)
    at org.liquibase.maven.plugins.AbstractLiquibaseMojo.execute (AbstractLiquibaseMojo.java:886)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:569)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/configuration/ConfiguredValueModifierFactory.java
+++ b/liquibase-standard/src/main/java/liquibase/configuration/ConfiguredValueModifierFactory.java
@@ -40,7 +40,7 @@ public class ConfiguredValueModifierFactory  implements SingletonObject {
         for (Iterator<ConfiguredValueModifier> iterator = allInstances.descendingIterator(); iterator.hasNext(); ) {
             ConfiguredValueModifier allInstance = iterator.next();
             String overriddenValue = allInstance.override(configuredValue);
-            if (!configuredValue.equals(overriddenValue)) {
+            if (configuredValue == null || !configuredValue.equals(overriddenValue)) {
                 return overriddenValue;
             }
         }

